Assets/Scripts/PlayerController.cs

```diff
@@ -1,27 +1,82 @@
-﻿using System.Collections;
-using System.Collections.Generic;
-using UnityEngine;
+﻿using UnityEngine;
 
 public class PlayerController : MonoBehaviour
 {
 
     [SerializeField]
     private const float MoveSpeed = 2f;
     private const float JumpForce = 5.2f;
-    private const float DistToGround = 1.0f;
+    private const float DistToGround = 0.5f;
 
     private Vector3 _forward;
     private Vector3 _right;
-
-    private bool _onGround = true;
+    private Vector3 _heading;
 
     private Rigidbody _rb;
+    public BoxCollider Bc;
+    private Vector3[] groundSkinVertices = new Vector3[40];
+    private Vector3[] forwardSkinVertices = new Vector3[16];
 
     // Use this for initialization
     private void Start()
     {
         _rb = GetComponent<Rigidbody>();
+        _heading = Vector3.forward;
+        groundSkinVertices[0] = new Vector3(Bc.bounds.extents.x - 0.05f, 0, Bc.bounds.extents.z - 0.05f);
+        groundSkinVertices[1] = new Vector3(Bc.bounds.extents.x - 0.05f, 0, -(Bc.bounds.extents.z - 0.05f));
+        groundSkinVertices[2] = new Vector3(-(Bc.bounds.extents.x - 0.05f), 0, -(Bc.bounds.extents.z - 0.05f));
+        groundSkinVertices[3] = new Vector3(-(Bc.bounds.extents.x - 0.05f), 0, Bc.bounds.extents.z - 0.05f);
```

Comment:
what does `0.05f` mean and is is the same idea as the `forwardSkinVertices` `0.01f`? Are we able to combine them into a  single variable?

Reply:
yup, same idea. don't know if we can combine them because we might want different values for them